pkg/sql/importer/import_pgdump.go

```diff
@@ -0,0 +1,581 @@
+					semaCtx := tree.MakeSemaContext()
+					if _, err := expr.TypeCheck(ctx, &semaCtx, nil /* desired */); err != nil {
+						// If the expression does not type check, it may be a case of using
+						// a column that does not exist yet in a setval call (as is the case
+						// of PGDUMP output from ogr2ogr). We're not interested in setval
+						// calls during schema reading so it is safe to ignore this for now.
+						if f := expr.Func.String(); pgerror.GetPGCode(err) == pgcode.UndefinedColumn && f == "setval" {
+							continue
+						}
+						return err
+					}
+					ov := expr.ResolvedOverload()
+					// Search for a SQLFn, which returns a SQL string to execute.
+					fn, ok := ov.SQLFn.(eval.SQLFnOverload)
+					if !ok {
+						err := errors.Errorf("unsupported function call: %s in stmt: %s",
+							expr.Func.String(), stmt.String())
+						return wrapErrorWithUnsupportedHint(err)
+					}
+					// Attempt to convert all func exprs to datums.
+					datums := make(tree.Datums, len(expr.Exprs))
+					for i, ex := range expr.Exprs {
+						d, ok := ex.(tree.Datum)
+						if !ok {
+							// We got something that wasn't a datum so we can't call the
+							// overload. Since this is a SQLFn and the user would have
+							// expected us to execute it, we have to error.
+							return errors.Errorf("unsupported statement: %s", stmt)
+						}
+						datums[i] = d
+					}
+					// Now that we have all of the datums, we can execute the overload.
+					// TODO: better comment
+					fnSQL, err := fn(ctx, evalCtx, datums)
+					if err != nil {
+						return err
+					}
+					// We have some sql. Parse and process it.
+					fnStmts, err := parser.Parse(fnSQL)
+					if err != nil {
+						return err
+					}
+					for _, fnStmt := range fnStmts {
+						switch ast := fnStmt.AST.(type) {
+						case *tree.AlterTable:
+							alterTableHandler := &PostgresDDLHandler{}
+							err := bufferDDLPostgresStatement(ctx, evalCtx, ast, p, parentID, alterTableHandler)
+							if err != nil {
+								return err
+							}
+							handler.BufferedDDLStmts = append(handler.BufferedDDLStmts, alterTableHandler.BufferedDDLStmts[0])
+						default:
+							// We only support ALTER statements returned from a SQLFn.
+							return wrapErrorWithUnsupportedHint(errors.Errorf("unsupported %T statement: %s", stmt, stmt))
+						}
+					}
+				default:
+					return wrapErrorWithUnsupportedHint(errors.Errorf("unsupported %T SELECT expr: %s", expr, expr))
+				}
+			}
+		default:
+			return wrapErrorWithUnsupportedHint(errors.Errorf("unsupported %T SELECT %s", sel, sel))
+		}
+	case *tree.DropTable:
+		names := stmt.Names
+
+		// If we find a table with the same name in the target DB we are importing
+		// into and same public schema, then we throw an error telling the user to
+		// drop the conflicting existing table to proceed.
+		// Otherwise, we silently ignore the drop statement and continue with the import.
```

Comment:
This is probably my ignorance showing but I find it odd that pgdump files would ever contain conflicting table names. If this is not expected, why not let the statements play out in the transaction and error there?